src/rpc/methods/eth.rs

```diff
@@ -179,6 +325,196 @@ impl HasLotusJson for BlockNumberOrHash {
+    pub transactions_root: Hash,
+    pub receipts_root: Hash,
+    pub logs_bloom: Bloom,
+    pub difficulty: Uint64,
+    pub total_difficulty: Uint64,
+    pub number: Uint64,
+    pub gas_limit: Uint64,
+    pub gas_used: Uint64,
+    pub timestamp: Uint64,
+    pub extra_data: Bytes,
+    pub mix_hash: Hash,
+    pub nonce: Nonce,
+    pub base_fee_per_gas: BigInt,
+    pub size: Uint64,
+    // can be Vec<Tx> or Vec<String> depending on query params
+    pub transactions: Transactions,
+    pub uncles: Vec<Hash>,
+}
+
+impl Block {
+    pub fn new() -> Self {
+        Self {
+            gas_limit: Uint64(BLOCK_GAS_LIMIT),
+            logs_bloom: Bloom(ethereum_types::Bloom(FULL_BLOOM)),
+            sha3_uncles: Hash(ethereum_types::H256::from_str(EMPTY_UNCLES).unwrap()),
+            ..Default::default()
+        }
+    }
+}
+
+lotus_json_with_self!(Block);
+
+#[derive(PartialEq, Debug, Clone, Default, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Tx {
+    pub chain_id: Uint64,
+    pub nonce: Uint64,
+    pub hash: Hash,
+    pub block_hash: Hash,
+    pub block_number: Uint64,
+    pub transaction_index: Uint64,
+    pub from: Address,
+    #[serde(skip_serializing_if = "LotusJson::is_none", default)]
+    pub to: LotusJson<Option<Address>>,
+    pub value: BigInt,
+    pub r#type: Uint64,
+    pub input: Bytes,
+    pub gas: Uint64,
+    pub max_fee_per_gas: BigInt,
+    pub max_priority_fee_per_gas: BigInt,
+    pub access_list: Vec<Hash>,
+    pub v: BigInt,
+    pub r: BigInt,
+    pub s: BigInt,
+}
+
+impl Tx {
+    pub fn eth_hash(&self) -> Hash {
+        let eth_tx_args: TxArgs = self.clone().into();
+        eth_tx_args.hash()
+    }
+}
+
+lotus_json_with_self!(Tx);
+
+#[derive(Debug, Clone, Default)]
+struct TxArgs {
+    pub chain_id: u64,
+    pub nonce: u64,
+    pub to: Option<Address>,
+    pub value: BigInt,
+    pub max_fee_per_gas: BigInt,
+    pub max_priority_fee_per_gas: BigInt,
+    pub gas_limit: u64,
+    pub input: Vec<u8>,
+    pub v: BigInt,
+    pub r: BigInt,
+    pub s: BigInt,
+}
+
+impl From<Tx> for TxArgs {
+    fn from(tx: Tx) -> Self {
+        Self {
+            chain_id: tx.chain_id.0,
+            nonce: tx.nonce.0,
+            to: tx.to.0,
+            value: tx.value,
+            max_fee_per_gas: tx.max_fee_per_gas,
+            max_priority_fee_per_gas: tx.max_priority_fee_per_gas,
+            gas_limit: tx.gas.0,
+            input: tx.input.0,
+            v: tx.v,
+            r: tx.r,
+            s: tx.s,
+        }
+    }
+}
+
+#[allow(clippy::indexing_slicing)]
+fn format_u64(value: u64) -> BytesMut {
+    if value != 0 {
+        let i = (value.leading_zeros() / 8) as usize;
+        let bytes = value.to_be_bytes();
+        bytes[i..].into()
+    } else {
+        // If all bytes are zero, return an empty slice
+        BytesMut::new()
+    }
+}
+
+#[allow(clippy::indexing_slicing)]
+fn format_bigint(value: &BigInt) -> BytesMut {
```

Comment:
Is it unit-tested?

Reply:
I've added a test. Turns out that `format_bigint` only works for unsigned bigints (but that was expected).
I will change the `eth::BigInt` constructor to reflect this.

Reply:
In go there's no `big.BigUInt` only `big.BigInt`. Why don't we use `BigUInt` on our side in `eth` module? Just a question.

Reply:
Likely nobody thought of using `BigUint`. :) Not sure if there won't be some wild `-1` numbers every now and then, I'd leave it to `BigInt` which is everywhere in our codebase anyway.

Reply:
Exactly.